### ISSUE TITLE
Handling of local parameters

### ISSUE THREAD

**Issue by LeonardSchmiester** [PEtab-dev/PEtab#337]
At the moment, local parameters cannot be directly estimated within PEtab, as this requires unique parameter Ids. One would need to make them global parameters.
Maybe instead we could use a unique naming scheme like in AMICI (${reaction_id}_${local_par_id}) to also allow for local parameters

**Comment by FFroehlich** [comment]
https://github.com/PEtab-dev/PEtab/blob/8b03c8f20c8042c5375adabfe23bab19c3a649b0/petab/sbml.py#L115 ?

**Comment by dweindl** [comment]
> https://github.com/PEtab-dev/PEtab/blob/8b03c8f20c8042c5375adabfe23bab19c3a649b0/petab/sbml.py#L115
> ?

Good catch. So far unused here. Legacy... And from before I knew that libsbml already implements that same functionality.

Question is mostly *if* we want to handle local parameters that way.

**Comment by JanHasenauer** [comment]
Is there a disadvantage with this and would there be another good option?

**Comment by dweindl** [comment]
> Is there a disadvantage with this and would there be another good option?

Disadvantage of allowing for local parameters: They don't necessarily have a unique ID. It's a per-reaction namespace. In PEtab we'd need globally identifiable parameter IDs. The common way is using ${reaction_id}_${local_par_id}, but this is still not necessarily unique.

**Comment by JanHasenauer** [comment]
Okay, but this is a problem of local parameters in general but not the suggested implementation. As we want to support available SBML models, we should in my opinion also support local parameters.

The common was sounds good and we should then be check for uniqueness.

**Comment by dweindl** [comment]
The changes proposed in #538 also provide a means to handle local parameters.

**Comment by matthiaskoenig** [comment]
libSBML has functionality for converting local to global parameters. I am using this by default on all models when reading them to get rid of any local parameters.
```python
doc: libsbml.SBMLDocument

# promote local parameters
props = libsbml.ConversionProperties()
props.addOption(
    "promoteLocalParameters", True, "Promotes all Local Parameters to Global ones"
)
if doc.convert(props) == libsbml.LIBSBML_OPERATION_SUCCESS:
    logger.info(f"Promotion of local parameters successful: {doc}")
else:
    logger.error(f"Promotion of local parameters failed: {doc}")
return doc
```
This is the standard way most tools are solving the local parameter issue.
